docs/contract.rst

```diff
@@ -0,0 +1,60 @@
+LinkableContract
+================
+
+`Py-EthPM` uses a custom subclass of ``Web3.contract.Contract`` to manage contract factories and instances which might require bytecode linking. To create a deployable contract factory, both the contract type's `abi` and `deployment_bytecode` must be available in the Package's manifest.
```

Comment:
@njgheorghita this might be worth incorperating into the core web3 API.

```python
w3 = Web3(...)
w3.eth.set_default_contract_factory_class(MyCustomContractClass)
c = w3.eth.contract(...)
assert isinstance(c, MyCustomContractClass)
```